app/javascript/components/pages/ApartmentShow.test.js

```diff
@@ -0,0 +1,67 @@
+import { render, screen } from "@testing-library/react";
+import ApartmentShow from "./ApartmentShow.js";
+import { MemoryRouter, Routes, Route } from "react-router-dom";
+import "@testing-library/jest-dom";
+import React from "react";
+import { isEqual } from "@rails/webpacker/package/utils/helpers.js";
+
+const apartments = [
+  {
+    bathrooms: 1.5,
+    address: "123 forth st.",
+    city: "AAron",
+    state: "SC",
+    bedrooms: 3,
+    price: 5000,
+    utilities: "gas and electric",
+    pets: true,
+    square_footage: 200,
+    image:
+      "https://images.unsplash.com/photo-1522708323590-d24dbb6b0267?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80",
+  },
+
+  {
+    bathrooms: 2,
+    address: "124 forth st.",
+    city: "AAron",
+    state: "SC",
+    bedrooms: 3,
+    price: 5560,
+    utilities: "gas and electric",
+    pets: true,
+    square_footage: 210,
+    image:
+      "https://images.unsplash.com/photo-1493809842364-78817add7ffb?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1170&q=80",
+  },
```

Comment:
Your mock apartments created for testing must reflect exactly how your database queries would look like. Meaning, your show page does a comparison of information that Rails handles but is absent in your test since you are working with mock data.